cypress/e2e/userProfile.cy.js

```diff
@@ -0,0 +1,31 @@
+import * as page from '../fixtures/URLs.json';
+
+describe('Edit user profile and check if the changes are visible after page reload', () => {
+  const username = `test_${Date.now()}`;
+  beforeEach(() => {
+    cy.login(Cypress.env('loginEmail'), Cypress.env('loginPassword'));
+  });
+
+  it('User can edit their profile username ', () => {
+    cy.visit(page.profile, { failOnStatusCode: false });
+    cy.get('button').contains('EDIT').click();
+    cy.get('input[name="username"]').clear().type(username);
+    cy.get('button').contains('EDIT PROFILE').click();
+    cy.reload();
+    cy.get('h2').should('contain.text', username);
+  });
+
+  it('User can edit their profile image', () => {
+    cy.visit(page.profile, { failOnStatusCode: false });
+    cy.get('button').contains('EDIT').click();
+    cy.get('input[type="file"]').should('exist');
+    cy.get('input[type="file"]').selectFile('cypress/fixtures/avatar.jpg', {
+      force: true,
+    });
+    cy.get('button').contains('EDIT PROFILE').click();
+    cy.reload();
+    cy.get('img[alt="User avatar"]')
+      .should('be.visible')
+      .and('have.attr', 'src');
```

Comment:
Great! I think this part of the test can be improved a bit. If we check that the `src` value contains a part of the url that we are enforcing in the code we can verify that it's in fact displaying an image from our bucket.

I played around with it a bit and came up with the line below that should do it for us:

```js
cy.get('img[alt="User avatar"]')
      .should('be.visible')
      .invoke('attr', 'src')
      .then((src) => {
        expect(src).to.match(/public\/images/gi);
      });
```